haystack/preview/document_stores/memory/_filters.py

```diff
@@ -224,7 +224,7 @@ def match(conditions: Any, document: Document, _current_key=None):
     if isinstance(conditions, dict):
         # Check for malformed filters, like {"name": {"year": "2020"}}
         if _current_key and any(key not in RESERVED_KEYS for key in conditions.keys()):
-            raise ValueError(
+            raise FilterError(
```

Comment:
Is there a specific reason why `FilterError` is used in some parts of this code while `MemoryDocumentStoreFilterError` is used in other parts?

Reply:
Ah no, that's a leftover, good catch!